projects/packages/forms/src/blocks/contact-form/edit.js

```diff
@@ -72,332 +72,299 @@ const PRIORITIZED_INSERTER_BLOCKS = [ ...map( validFields, block => `jetpack/${
-			) {
-				setIsPatternsModalOpen( true );
-			}
-			// eslint-disable-next-line react-hooks/exhaustive-deps
-		}, [] );
+		selectBlock( clientId );
+	};
 
-		const renderSubmissionSettings = () => {
-			return (
-				<>
-					<InspectorHint>
-						{ __( 'Customize the view after form submission:', 'jetpack-forms' ) }
-					</InspectorHint>
-					<SelectControl
-						label={ __( 'On Submission', 'jetpack-forms' ) }
-						value={ customThankyou }
-						options={ [
-							{ label: __( 'Show a summary of submitted fields', 'jetpack-forms' ), value: '' },
-							{ label: __( 'Show a custom text message', 'jetpack-forms' ), value: 'message' },
-							{ label: __( 'Redirect to another webpage', 'jetpack-forms' ), value: 'redirect' },
-						] }
-						onChange={ newMessage => setAttributes( { customThankyou: newMessage } ) }
-						__nextHasNoMarginBottom={ true }
-						__next40pxDefaultSize={ true }
-					/>
+	useEffect( () => {
+		// Populate default variation on older versions of WP or GB that don't support variations.
+		if ( ! hasInnerBlocks && ! registerBlockVariation ) {
+			setVariation( defaultVariations[ 0 ] );
+		}
+	} );
 
-					{ 'redirect' !== customThankyou && (
-						<TextControl
-							label={ __( 'Message Heading', 'jetpack-forms' ) }
-							value={ customThankyouHeading }
-							placeholder={ __( 'Your message has been sent', 'jetpack-forms' ) }
-							onChange={ newHeading => setAttributes( { customThankyouHeading: newHeading } ) }
-							__nextHasNoMarginBottom={ true }
-							__next40pxDefaultSize={ true }
-						/>
-					) }
+	useEffect( () => {
+		if (
+			! hasInnerBlocks &&
+			registerBlockVariation &&
+			! isPatternsModalOpen &&
+			window.location.search.indexOf( 'showJetpackFormsPatterns' ) !== -1
+		) {
+			setIsPatternsModalOpen( true );
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
 
-					{ 'message' === customThankyou && (
-						<TextareaControl
-							label={ __( 'Message Text', 'jetpack-forms' ) }
-							value={ customThankyouMessage }
-							placeholder={ __( 'Thank you for your submission!', 'jetpack-forms' ) }
-							onChange={ newMessage => setAttributes( { customThankyouMessage: newMessage } ) }
-							__nextHasNoMarginBottom={ true }
-						/>
-					) }
+	let elt;
 
-					{ 'redirect' === customThankyou && (
-						<div>
-							<URLInput
-								label={ __( 'Redirect Address', 'jetpack-forms' ) }
-								value={ customThankyouRedirect }
-								className="jetpack-contact-form__thankyou-redirect-url"
-								onChange={ newURL => setAttributes( { customThankyouRedirect: newURL } ) }
-							/>
-						</div>
-					) }
-				</>
+	if ( ! isModuleActive ) {
+		if ( isLoadingModules ) {
+			elt = <ContactFormSkeletonLoader />;
+		} else {
+			elt = (
+				<ContactFormPlaceholder
+					changeStatus={ changeStatus }
+					isModuleActive={ isModuleActive }
+					isLoading={ isChangingStatus }
+				/>
 			);
-		};
-
-		const renderVariationPicker = () => {
```

Comment:
Again, this was only used in one place, so it's moved inline.

Reply:
I think it would be even better to make this a separate component too. Half of the calls to store are for the picker.